src/task/SubscribeAndReceiveSubscribeAckTask.ts

```diff
@@ -94,6 +99,59 @@ export default class SubscribeAndReceiveSubscribeAckTask extends BaseTask {
     this.context.videoStreamIndex.integrateSubscribeAckFrame(subscribeAckFrame);
   }
 
+  // Our backends currently expect the video subscriptions passed in subscribe to precisely
+  // line up with the media sections, with a zero for any video send or inactive section.
+  //
+  // Firefox occasionally tosses stopped transceivers at the end of the SDP without reason
+  // and in general we don't want to be at the mercy of SDP sections not being in the same
+  // order as `getTransceivers`, so we simply recalculate the array here to enforce that
+  // expected invarient until we refactor our signaling to simply take a mapping of MID to
+  // subscription.
+  //
+  // This only works on Unified Plan SDPs
+  private fixUpSubscriptionOrder(sdp: string, videoSubscriptions: number[]): number[] {
+    if (this.context.transceiverController.getMidForStreamId === undefined) {
```

Comment:
@xuesichao an example of why the required interface/default implementation is annoying :D. Do we really want to support custom transceiver controllers? not really for reasons like this....